Initial PR for DataBox

kushalvms opened the pull request:
1. Added tests for all the apis.
2. Added recorded sessions for all the tests.

REST specs PR: https://github.com/Azure/azure-rest-api-specs/pull/2772

<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
<!--
Please add an informative description that covers that changes made by the pull request.

If you are regenerating your SDK based off of a new swagger spec, please add the link to the corresponding swagger spec pull request that has been merged in the azure-rest-api-specs repository
-->

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

kushalvms committed:
Initial PR for DataBox

1. Added tests for all the apis.
2. Added recorded sessions for all the tests.

kushalvms committed:
OAV validation fixes

Fixed OAV issues in specs so updating same here

kushalvms committed:
Merge remote-tracking branch 'refs/remotes/Azure/psSdkJson6' into psSdkJson6

shahabhijeet commented on src/SDKs/DataBox/DataBox.Tests/DataBox.Tests.csproj:
```diff
@@ -0,0 +1,37 @@
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Update="Microsoft.NET.Test.Sdk" Version="15.6.2" />
```
@kushalvms can you remove these references. These will be added implicitly to your projects and we would like it to be that way, in order to upgrade common packages.

kushalvms replied:
removed

kushalvms replied:
removed

shahabhijeet commented on src/SDKs/DataBox/DataBox.Tests/DataBox.Tests.csproj:
```diff
@@ -0,0 +1,37 @@
+    <PackageReference Update="Microsoft.Rest.ClientRuntime.Azure.TestFramework" Version="1.7.2" />
+  </ItemGroup>
+  <ItemGroup>
+    <Folder Include="SessionRecords\" />
```
@kushalvms this is not needed. Removing it from csproj will still add the required directory to your projects

kushalvms replied:
removed

shahabhijeet commented on src/SDKs/DataBox/Management.DataBox/Microsoft.Azure.Management.DataBox.csproj:
```diff
@@ -0,0 +1,26 @@
+		<PackageId>Microsoft.Azure.Management.DataBox</PackageId>
+		<Description>Microsoft Azure Management DataBox library</Description>
+		<AssemblyName>Microsoft.Azure.Management.DataBox</AssemblyName>
+		<Version>1.0.0-preview</Version>
```
@kushalvms if it's your first version, we recommend you start your version from 0.x.x-preview and all first preview pacakges start at 0.9.0-preview

shahabhijeet commented on src/SDKs/DataBox/Management.DataBox/Microsoft.Azure.Management.DataBox.csproj:
```diff
@@ -0,0 +1,26 @@
+	<PropertyGroup>
+		<TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+		<RootNamespace>Microsoft.Azure.Management.DataBox</RootNamespace>
+		<GeneratePackageOnBuild>true</GeneratePackageOnBuild>
```
@kushalvms can you remove this. We do nto want to create pacakges on build. We have separate targets for creating pakcages, this has direct impact on CI perf.
msbuild build.proj /t:CreateNugetPackage /p:Scope=SDKs\DataBox

shahabhijeet commented on src/SDKs/DataBox/Management.DataBox/Microsoft.Azure.Management.DataBox.csproj:
```diff
@@ -0,0 +1,26 @@
+		<AssemblyName>Microsoft.Azure.Management.DataBox</AssemblyName>
+		<Version>1.0.0-preview</Version>
+		<PackageTags>Microsoft Azure DataBox management;DataBox;DataBox management;</PackageTags>
+		<NoWarn>$(NoWarn);CS1591</NoWarn>
```
@kushalvms is this warning coming from your project? if yes we are asking help of all RPs to take care of their own project and not to introduce this warning.

shahabhijeet commented on src/SDKs/DataBox/Management.DataBox/Microsoft.Azure.Management.DataBox.csproj:
```diff
@@ -0,0 +1,26 @@
+		<Description>Microsoft Azure Management DataBox library</Description>
+		<AssemblyName>Microsoft.Azure.Management.DataBox</AssemblyName>
+		<Version>1.0.0-preview</Version>
+		<PackageTags>Microsoft Azure DataBox management;DataBox;DataBox management;</PackageTags>
```
@kushalvms all tags with spaces become individual tags on nuget.org
Suggestion:
I would recommend to limit your tags to limited ones that your customer is going to use.
For e.g. if someone wants to search for databox, they are not going to start their search by writing Azure. Rather use AzureDatabox as 1 word.

shahabhijeet commented on src/SDKs/DataBox/Management.DataBox/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,19 @@
+[assembly: AssemblyTitle("Microsoft Azure DataBox Management Library")]
+[assembly: AssemblyDescription("Provides Microsoft Azure DataBox management functions for managing the Microsoft Azure DataBox service.")]
+
+[assembly: AssemblyVersion("1.0.0.0")]
```
make it 0.9.0 for both Assembly and AssemblyFileVersion

shahabhijeet commented on src/SDKs/DataBox/Management.DataBox/generate.cmd:
```diff
@@ -0,0 +1,7 @@
+::
+
+@echo off
+call %~dp0..\..\..\..\tools\generate.cmd databox/resource-manager %*
```
@kushalvms we moving to .ps1 files form .cmd files so that we can easily add more commandline switches.
Can you model your ps1 from one of the existing one
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Compute/Management.Compute/generate.ps1

kushalvms replied:
added generate.ps1

shahabhijeet commented on src/SDKs/DataBox/AzSdk.RP.props:
```diff
@@ -0,0 +1,7 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--This file and it's contents are updated at build time moving or editing might result in build failure. Take due deligence while editing this file-->
+  <PropertyGroup>
+    <AzureApiTag />
```
@kushalvms we have a bug currently that will not update the meta data.
Can you build your project from command line once
msbuild build.proj /t:Build /p:Scope=SDKs\DataBox

shahabhijeet replied:
@kushalvms another thing you will have to do is get latest from upstream and generate using generate.ps1
This will create all the necessary meta data required for the PR to be merged in.

kushalvms replied:
locally the code generation is failing with latest autorest package version: 2.0.4262. I tried with the last version: 2.0.4215 and it passed. What am I missing here or is there any problem with the latest version?

kushalvms committed:
Added review comments

Addressed review comments
Moved to new generate.ps1

kushalvms committed:
Removing internal api

Removing internal api which is only being used by the runner service and
will always be behind feature in ARM.

kushalvms committed:
Merge remote-tracking branch 'refs/remotes/Azure/psSdkJson6' into psSdkJson6

kushalvms committed:
Removing the example file

Removing the example file of the internal api

kushalvms committed:
Removing test case also for the internal api

Removing test case also for the internal api

kushalvms commented:
specs got merged to Azure, please review.

dsgouda commented on src/SDKs/_metadata/databox_resource-manager.txt:
```diff
@@ -0,0 +1,5 @@
+﻿Executing AutoRest command
+cmd.exe /c autorest.cmd https://github.com/AzureDataBox/azure-rest-api-specs/blob/master/specification/databox/resource-manager/readme.md --csharp --csharp-sdks-folder=G:\AzureDataBox\azure-sdk-for-net\tools\..\src\SDKs\ --version=2.0.4215 --reflect-api-versions
+
+AutoRest code generation for https://github.com/AzureDataBox/azure-rest-api-specs/blob/master/specification/databox/resource-manager/readme.md failed. Please try again
```
Looks like the code generation failed here, please regenerate the code and update PR

dsgouda reviewed (changes requested):
Please generate the code using generate.cmd or generate.ps1 and update the PR

kushalvms committed:
Merge remote-tracking branch 'refs/remotes/Azure/psSdkJson6' into psSdkJson6

kushalvms committed:
Merge remote-tracking branch 'refs/remotes/Azure/psSdkJson6' into psSdkJson6

kushalvms committed:
Generated code with latest autorest

Generated code again with the latest autorest after specs merge
